_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,519 @@
+---
+layout: default
+title: Search with k-NN filters
+nav_order: 10
+parent: k-NN
+has_children: false
+has_math: true
+---
+
+# Search for k-NN with filters
+Introduced 2.4
+{: .label .label-purple }
+
+You can create custom filters using Query DSL search options to refine your k-NN searches. You define the filter criteria within the `knn_vector` field's `filter` subsection in your query. You can use all available Query DSL query types such as: term, range, regex and wildcard. To include or exclude results, you specify Boolean query clauses. You also specify a query point with the `knn_vector` type and search for nearest neighbors that match your filter criteria.
```

Comment:
This looks more like exhaustive list of query types, I think we need to highlight that those four types are only examples, although probably they are ones used the most. I don't think we need to list all types as there are way too many, and also it can be something custom that we're can't predict, like 3-rd party plugin. 
```
You can use all available Query DSL query types such as: term, range, regex and wildcard
```